realsense2_camera/src/base_realsense_node.cpp

```diff
@@ -785,6 +803,59 @@ void BaseRealSenseNode::publishPointCloud(rs2::points pc, const rclcpp::Time& t,
     _pc_filter->Publish(pc, t, frameset, frame_id);
 }
 
+bool BaseRealSenseNode::isSCStream(const stream_index_pair& sip)
+{
+    const rs2_stream stream = sip.first;
+    return (stream == RS2_STREAM_SAFETY || stream == RS2_STREAM_OCCUPANCY || stream == RS2_STREAM_LABELED_POINT_CLOUD);
+}
+
+void BaseRealSenseNode::publishLabeledPointCloud(rs2::labeled_points pc, const rclcpp::Time& t){
+    if(0 == _labeled_pointcloud_publisher->get_subscription_count())
+        return;
+    
+    ROS_DEBUG("Publishing Labeled Point Cloud Frame");
+
+    // Create the PointCloud message
+    sensor_msgs::msg::PointCloud2::UniquePtr msg_pointcloud = std::make_unique<sensor_msgs::msg::PointCloud2>();
+
+    // Define the fields of the PointCloud message
+    sensor_msgs::PointCloud2Modifier modifier(*msg_pointcloud);
+
+    modifier.setPointCloud2Fields(4, "x", 1, sensor_msgs::msg::PointField::FLOAT32,
+                                "y", 1, sensor_msgs::msg::PointField::FLOAT32,
+                                "z", 1, sensor_msgs::msg::PointField::FLOAT32,
+                                "label", 1, sensor_msgs::msg::PointField::UINT8);
+    modifier.resize(pc.size());
+
+    // Fill the PointCloud message with data
+    sensor_msgs::PointCloud2Iterator<float> iter_x(*msg_pointcloud, "x");
+    sensor_msgs::PointCloud2Iterator<float> iter_y(*msg_pointcloud, "y");
+    sensor_msgs::PointCloud2Iterator<float> iter_z(*msg_pointcloud, "z");
+    sensor_msgs::PointCloud2Iterator<float> iter_label(*msg_pointcloud, "label");
+    const rs2::vertex* vertex = pc.get_vertices();
+    const uint8_t* label = pc.get_labels();
+    msg_pointcloud->width = 320;
```

Comment:
@remibettan 
should we add get_width / get_height for rs2::labeleld_points? 
because in the future, labeleld point cloud might support different heights/widths/resolutions

Reply:
I understand we are getting unordered vertices, not an ordered ones.
But, its better to wait for @benlev inputs here, since he might want this meta info in the msg itself.

Reply:
This was a valid question :)
LPC resolution is now changed and I am phasing the same question..